go-controller/pkg/allocator/pod/pod_annotation_test.go

```diff
@@ -161,11 +163,9 @@ func Test_allocatePodAnnotationWithRollback(t *testing.T) {
 			},
 			wantUpdatedPod: true,
 			wantPodAnnotation: &util.PodAnnotation{
-				IPs:  ovntest.MustParseIPNets("192.168.0.4/24"),
-				MAC:  util.IPAddrToHWAddr(ovntest.MustParseIPNets("192.168.0.4/24")[0].IP),
-				Role: types.NetworkRolePrimary,
+				IPs: ovntest.MustParseIPNets("192.168.0.4/24"),
+				MAC: util.IPAddrToHWAddr(ovntest.MustParseIPNets("192.168.0.4/24")[0].IP),
 			},
-			role: types.NetworkRolePrimary,
```

Comment:
why did you remove the primary role from this test ?

Reply:
The test is "expect requested static IP, no gateway, no IPAM" we don't have primaries without gateway, the primary role here feels like copy paste and make testing this feature more difficult.